packages/synapse-react-client/src/components/QueryWrapper/ViewMoreQueryResultsButton.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect } from 'react'
+import { Box } from '@mui/material'
+import WideButton from '../styled/WideButton'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+import { useQueryContext } from '../QueryContext'
+import { useSuspenseQuery } from '@tanstack/react-query'
+
+export type ViewMoreQueryResultsButtonProps = {
+  hasNextPage: boolean
+  fetchNextPage: () => void
+  isFetchingNextPage: boolean
+  initialLimit?: number
+  initialLimitIsApplied?: boolean
+  onRemoveInitialLimit?: () => void
+}
+
+export function ViewMoreQueryResultsButton(
```

Comment:
New component that handles infinite pagination. Logic was based on previous behavior in `CardContainer`